### PR TITLE
Add security subcategories and bump rubocop to 0.80.0

### PR DESCRIPTION
**Pull request by josemiguelmelo**

### CODE REVIEW

File: src/main/ruby/codacy/rubocop/generator.rb [hunk 1]
```diff
@@ -105,8 +119,9 @@ def self.run(file_path = "rubocop-doc.yml")
             patternId:  cop_data[:name].gsub("/", "_"),
             level:      level(cop_data),
             category:   category(cop_data),
+            subcategory: subcategory(cop_data),
             parameters: parameters(cop_data)
-          }
+          }.delete_if { |key, value| value == nil }
```
Comment (rtfpessoa) [inline src/main/ruby/codacy/rubocop/generator.rb]: Codacy Issue found: [Unused block argument - `key`. If it's necessary, use `_` or `_key` as an argument name to indicate that it won't be used.](https://app.codacy.com/gh/codacy/codacy-rubocop/pullRequest?prid=5072804)

File: src/main/ruby/codacy/rubocop/generator.rb [hunk 2]
```diff
@@ -92,6 +92,20 @@ def self.category(cop_data)
...
+      def self.subcategory(cop_data)
+        if category(cop_data) == "Security"
+          case cop_data[:name] 
+          when "Security/JSONLoad", "Security/MarshalLoad"
+            "InsecureModulesLibraries"
+          when "Security/Open"
+            "CommandInjection"
+          when "Security/YAMLLoad", "Security/Eval"
+            "InputValidation"
+          else nil
```
Comment (rtfpessoa) [inline src/main/ruby/codacy/rubocop/generator.rb]: Codacy Issue found: [Redundant `else`-clause.](https://app.codacy.com/gh/codacy/codacy-rubocop/pullRequest?prid=5072804)

File: src/main/ruby/codacy/rubocop/generator.rb (hunk 1 above)
Comment (rtfpessoa) [inline src/main/ruby/codacy/rubocop/generator.rb]: Codacy Issue found: [Prefer the use of the `nil?` predicate.](https://app.codacy.com/gh/codacy/codacy-rubocop/pullRequest?prid=5072804)

File: src/main/ruby/codacy/rubocop/generator.rb [hunk 3]
```diff
@@ -105,12 +125,13 @@ def self.run(file_path = "rubocop-doc.yml")
...
             category:   category(cop_data),
+            subcategory: subcategory(cop_data),
             parameters: parameters(cop_data)
-          }
+          }.delete_if { |_, value| value.nil? }
         end
         data      = {
           name:     "Rubocop",
-          version:  File.read(".rubocop-version").strip,
+          version:  Gem.loaded_specs["rubocop"].version, #File.read(".rubocop-version").strip,
```
Comment (franciscodua) [inline src/main/ruby/codacy/rubocop/generator.rb]: Reminder: Remove this comment

Comment (josemiguelmelo) [reply src/main/ruby/codacy/rubocop/generator.rb]: Nice catch! Removed it ✅

File: src/main/ruby/codacy/rubocop/generator.rb [hunk 4]
```diff
@@ -40,12 +40,19 @@ def self.generate_json_file(hash)
...
+        while description.length() > 254 do
+          description = description.slice(0..(description.rindex('.')-1))
+        end
+        description
+      end
+
       def self.run(file_path = "rubocop-doc.yml")
         cops_data    = YAML.load_file(file_path)
-        descriptions = cops_data.map do |cop_data|
+        descriptions = cops_data.map do |cop_data|
```
[the remark this answers is not in the document]
Comment (josemiguelmelo) [reply src/main/ruby/codacy/rubocop/generator.rb]: Removed it ✅

File: src/main/ruby/codacy/rubocop/generator.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (josemiguelmelo) [reply src/main/ruby/codacy/rubocop/generator.rb]: Already removed it ✅

**Pull request merged**